azure-iot-device/tests/provisioning/models/test_registration_result.py

```diff
@@ -53,27 +44,53 @@ def test_registration_result_instantiated_correctly(self):
         assert registration_result.registration_state.last_update_date_time == fake_last_update_dttm
         assert registration_result.registration_state.etag == fake_etag
 
+    @pytest.mark.it("Has a to string representation composed of registration state and status")
+    def test_registration_result_to_string(self):
+        fake_registration_state = create_registraion_state()
+        registration_result = create_registration_result(fake_registration_state)
+
+        string_repr = "\n".join([str(fake_registration_state), fake_status])
+        assert str(registration_result) == string_repr
+
     @pytest.mark.it("Has attributes that do not have setter")
     def test_some_properties_of_result_are_not_settable(self):
-        registration_result = RegistrationResult("RequestId123", "Operation456", "emitted", None)
+        registration_result = create_registration_result()
         with pytest.raises(AttributeError, match="can't set attribute"):
-            registration_result.request_id = "MyNimbus2000"
+            registration_result.request_id = "RequestId123"
             registration_result.operation_id = "WhompingWillow"
-            registration_result.status = "Flying"
+            registration_result.status = "Apparating"
             registration_result.registration_state = "FakeRegistrationState"
 
     @pytest.mark.it("Has `RegistrationState` with properties that do not have setter")
     def test_some_properties_of_state_are_not_settable(self):
-        registration_state = RegistrationState(
-            fake_device_id,
-            fake_assigned_hub,
-            fake_sub_status,
-            fake_created_dttm,
-            fake_last_update_dttm,
-            fake_etag,
-        )
+        registration_state = create_registraion_state()
+
         with pytest.raises(AttributeError, match="can't set attribute"):
-            registration_state.device_id = fake_device_id
-            registration_state.assigned_hub = fake_assigned_hub
-            registration_state.sub_status = fake_sub_status
-            registration_state.created_date_time = fake_created_dttm
+            registration_state.device_id = "Thunderbolt"
+            registration_state.assigned_hub = "MinistryOfMagic"
+            registration_state.sub_status = "Apparating"
+            registration_state.created_date_time = datetime.datetime(3000, 10, 17)
+
+    @pytest.mark.it(
+        "Has a to string representation composed of device id, assigned hub and sub status"
+    )
+    def test_registration_state_to_string(self):
+        registration_state = create_registraion_state()
+
+        string_repr = "\n".join([fake_device_id, fake_assigned_hub, fake_sub_status])
+        assert str(registration_state) == string_repr
+
+
+def create_registraion_state():
```

Comment:
This should be a fixture

Reply:
I did not make this a fixture yet. Though this can be dome but I am in favor of instantiating here..

---
In reply to: [315814200](https://github.com/Azure/azure-iot-sdk-python-preview/pull/169#discussion_r315814200) [](ancestors = 315814200)